connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -188,11 +196,52 @@ public ConfigDef config() {
         return MirrorSourceConfig.CONNECTOR_CONFIG_DEF;
     }
 
+    @Override
+    public org.apache.kafka.common.config.Config validate(Map<String, String> props) {
+        List<ConfigValue> configValues = MirrorSourceConfig.CONNECTOR_CONFIG_DEF.validate(props);
+        if ("required".equals(props.get(EXACTLY_ONCE_SUPPORT_CONFIG))) {
+            if (!consumerUsesReadCommitted(props)) {
+                ConfigValue exactlyOnceSupport = configValues.stream()
+                        .filter(cv -> EXACTLY_ONCE_SUPPORT_CONFIG.equals(cv.name()))
+                        .findAny()
+                        .orElseGet(() -> {
+                            ConfigValue result = new ConfigValue(EXACTLY_ONCE_SUPPORT_CONFIG);
+                            configValues.add(result);
+                            return result;
+                        });
+                // The Connect framework will already generate an error for this property if we return ExactlyOnceSupport.UNSUPPORTED
+                // from our exactlyOnceSupport method, but it will be fairly generic
+                // We add a second error message here to give users more insight into why this specific connector can't support exactly-once
+                // guarantees with the given configuration
+                exactlyOnceSupport.addErrorMessage(
+                        "MirrorSourceConnector can only provide exactly-once guarantees when its source consumer is configured with "
+                                + ConsumerConfig.ISOLATION_LEVEL_CONFIG + " set to '" + READ_COMMITTED + "'; "
+                                + "otherwise, records from aborted and uncommitted transactions will be replicated from the "
+                                + "source cluster to the target cluster."
+                );
+            }
+        }
+        return new org.apache.kafka.common.config.Config(configValues);
+    }
+
     @Override
     public String version() {
         return AppInfoParser.getVersion();
     }
 
+    @Override
+    public ExactlyOnceSupport exactlyOnceSupport(Map<String, String> props) {
+        return consumerUsesReadCommitted(props)
+                ? ExactlyOnceSupport.SUPPORTED
+                : ExactlyOnceSupport.UNSUPPORTED;
+    }
+
+    private boolean consumerUsesReadCommitted(Map<String, String> props) {
+        MirrorSourceConfig config = new MirrorSourceConfig(props);
```

Comment:
I think the issue is that if the configs contain a bad value this will throw. So when called from `validate()` we lose the error messages computed by `MirrorSourceConfig.CONNECTOR_CONFIG_DEF.validate(props);` and just throw `ConfigException`. 

You can reproduce if you add:
```
props.put(OFFSET_LAG_MAX, "bad");
configValue = validateProperty(OFFSET_LAG_MAX, props);
assertTrue(configValue.isPresent());
errorMessages = configValue.get().errorMessages();
assertEquals(1, errorMessages.size());
errorMessage = errorMessages.get(0);
assertTrue(
    errorMessages.get(0).contains(OFFSET_LAG_MAX),
    "Error message \"" + errorMessage + "\" should have mentioned the 'offset.lag.max' property"
);
```
to `MirrorSourceConnectorTest.testExactlyOnceSupportValidation()`

Reply:
This turned out to be much trickier to address than expected, but I've made this method impervious to unrelated invalid properties.

This required being able to determine the full config used for the source consumer without constructing a `MirrorSourceConfig` instance, which in turn required (or at least, was facilitated by) pulling out the find-all-entries-with-prefix logic in `AbstractConfig::originalsWithPrefix` into a static method in `Utils`.

I couldn't think of a less invasive way to accomplish this, but it does seem worth the tradeoff; LMKWYT.